### PR TITLE
Implement FetchRequest repository with entity conversion

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Implement FetchRequest repository with entity conversion

**Related issue(s)**
#226

### CODE REVIEW

File: components/director/internal/domain/fetchrequest/converter_test.go [hunk 1]
```diff
@@ -97,3 +101,145 @@ func TestConverter_InputFromGraphQL(t *testing.T) {
...
+		ExpectedErrMessage string
+	}{
+		{
+			Name:               "All properties given",
+			Input:              fixFetchRequestEntity(t, "1", timestamp),
+			Expected:           fixFetchRequestModel("1", timestamp),
+			ExpectedErrMessage: "",
+		},
+		{
+			Name: "Empty value",
```
Comment (dbadura) [inline components/director/internal/domain/fetchrequest/converter_test.go]: `returns error when ...`

File: components/director/internal/domain/fetchrequest/converter_test.go [hunk 2]
```diff
@@ -97,3 +101,145 @@ func TestConverter_InputFromGraphQL(t *testing.T) {
...
+					Timestamp: timestamp,
+					Condition: model.FetchRequestStatusConditionFailed,
+				},
+			},
+			ExpectedErrMessage: "",
+		},
+		{
+			Name: "Error",
+			Input: fetchrequest.Entity{
+				Auth: sql.NullString{
```
Comment (dbadura) [inline components/director/internal/domain/fetchrequest/converter_test.go]: Maybe is worth to create function similar to `newNullableString` from `repo` package, but string as parameter.

File: components/director/internal/domain/fetchrequest/repository.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/repo"
+	"github.com/pkg/errors"
+)
+
+const fetchRequestTable string = `public.fetch_requests`
+
+var tableColumns = []string{"id", "tenant_id", "api_def_id", "event_api_def_id", "document_id", "url", "auth", "mode", "filter", "status_condition", "status_timestamp"}
```
Comment (dbadura) [inline components/director/internal/domain/fetchrequest/repository.go]: `fetchRequestColumns`? Acoording to `runtime` repository impl.?

Comment (pkosiec) [reply components/director/internal/domain/fetchrequest/repository.go]: I want it more generic, `tableColumns` sounds better for me

File: components/director/internal/domain/fetchrequest/repository_test.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+		ctx := persistence.SaveToContext(context.TODO(), db)
+		repo := fetchrequest.NewRepository(mockConverter)
+		// WHEN
+		err := repo.Create(ctx, &frModel)
+		// THEN
+		require.NoError(t, err)
+	})
+
+	t.Run("DB Error", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply components/director/internal/domain/fetchrequest/repository_test.go]: Firstly we need to come up with single convention of naming tests - but I would agree with being consistent in this topic.

File: components/director/internal/domain/fetchrequest/fixtures_test.go [hunk 5]
```diff
@@ -62,3 +66,103 @@ func fixGQLFetchRequestInput(url, filter string) *graphql.FetchRequestInput {
...
+			String: filter,
+			Valid:  true,
+		},
+		StatusCondition: string(model.FetchRequestStatusConditionSucceeded),
+		StatusTimestamp: timestamp,
+		Auth: sql.NullString{
+			Valid:  true,
+			String: string(bytes),
+		},
+		APIDefID:      sql.NullString{},
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply components/director/internal/domain/fetchrequest/fixtures_test.go]: It is empty value, so I would keep it as it is

**Pull request merged**